### PR TITLE
Bulk upgrade test case #1230

### PR DESCRIPTION
**Pull request by omgoswami**
This PR adds a test case that automatically checks that when a Helm chart in a ResourceComposition is updated, all running application instances also get upgraded. This was done by adapting the steps listed in [https://github.com/cloud-ark/kubeplus/blob/master/examples/multitenancy/managed-service/appday2ops/steps.txt](url)

### CODE REVIEW

File: tests/tests.py [hunk 1]
```diff
@@ -136,6 +137,179 @@ def test_application_update(self):
         cmd = "kubectl delete -f ../examples/multitenancy/hello-world/hello-world-service-composition-localchart.yaml --kubeconfig=../kubeplus-saas-provider.json"
         out, err = TestKubePlus.run_command(cmd)
 
+    def test_application_upgrade(self):
+        # imports
+        from kubernetes import config
```
Comment (devdattakulkarni) [inline tests/tests.py:142]: You can move the imports at the top.

File: tests/tests.py [hunk 2]
```diff
@@ -136,6 +137,179 @@ def test_application_update(self):
...
+            else:
+                print(f"Port {port} has the following error: {error}")
+            http.close()
+            return response.decode('utf-8')
+        
+        # get number of users from HTML response
+        def count_users(response):
+            num_users = 0
+            if "No users" not in response:
+                soup = BeautifulSoup(response, 'lxml')
```
Comment (devdattakulkarni) [inline tests/tests.py:183]: Do we need to include any library in requirements.txt for BeautifulSoup?

**Comment by devdattakulkarni** [comment]
Overall, looks good.

I have added two in-line comments.

Also, please remove __pycache__ folder from the commit.

**Comment by devdattakulkarni** [comment]
@omgoswami ^^

File: tests/tests.py (hunk 1 above)
Comment (omgoswami) [reply tests/tests.py:142]: Ok!

File: tests/tests.py (hunk 2 above)
Comment (omgoswami) [reply tests/tests.py:183]: Yes, I added bs4 (and the dependent lxml module) and kubernetes to the requirements.txt just now.

**Comment by omgoswami** [comment]
Incorporated your suggestions!

**Comment by devdattakulkarni** [comment]
@omgoswami ^^

**Comment by devdattakulkarni** [comment]
@omgoswami Tests are passing!
Merging the PR now.

**Pull request merged**